src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,408 @@
+          end
+        end
+
+        clipped = clipped.left
+        if clipped.left == startclipped.left
+          break
+        end
+      end
+    end
+
+    clipping = clipping.left
+    if clipping.left == startclipping.left
+      break
+    end
+  end
+
+  collectedrings::typeof(polyrings) = collectclipped(entering)
+
+  # When no interesction have been registered with any of the rings, the clipping ring either
+  # encircles everything or is completely contained in the clipped polygon.
+  if all(isequal(false), intersected)
+    o = orientation(ring)
+    contained = [v ∈ PolyArea(ring) for v in vertices(polyrings[1])]
+    if (o == CCW && all(contained))
+      return poly
+    end
+    if (o == CW && !all(contained))
+      push!(collectedrings, polyrings[1])
+      intersected[1] = true
+      for poly_ring in polyrings[2:end]
+        if !all([v ∈ PolyArea(ring) for v in vertices(poly_ring)])
+          push!(collectedrings, poly_ring)
+        end
+      end
+      if any([v ∈ PolyArea(polyrings[1]) for v in vertices(ring)])
+        push!(collectedrings, ring)
+      end
+      return PolyArea(collectedrings)
+    end
+    if all([v ∈ PolyArea(polyrings[1]) for v in vertices(ring)])
+      push!(collectedrings, ring)
+    end
+  end
+
+  # Handle all the non-intersected rings. Add them if they are contained in the clipping ring.
+  polys::Vector{PolyArea} = []
+  for r in collectedrings
+    newpolyrings = [r]
+    valid = true
+    for k in eachindex(intersected)
+      if !intersected[k]
+        if orientation(ring) == CCW
+          # Check if majority of vertices are inside the clipping ring.
+          ins = count(isequal(true), [v ∈ PolyArea(ring) for v in vertices(polyrings[k])])
+          if ins > (length(vertices(polyrings[k])) - ins)
+            if orientation(polyrings[k]) == CCW
+              pushfirst!(newpolyrings, polyrings[k])
+            else
+              push!(newpolyrings, polyrings[k])
+            end
+            intersected[k] = true
+          elseif vertices(ring)[1] ∈ PolyArea(polyrings[k]) && k > 1
+            # If the ring is contained within one of the inner rings, invalidate it.
+            valid = false
+          end
+        else
+          # Check if majority of vertices are outside the clipping ring.
+          ins = count(isequal(true), [v ∈ PolyArea(ring) for v in vertices(polyrings[k])])
+          if ins < (length(vertices(polyrings[k])) - ins)
+            if orientation(polyrings[k]) == CCW
+              pushfirst!(newpolyrings, polyrings[k])
+            else
+              push!(newpolyrings, polyrings[k])
+            end
+            intersected[k] = true
+          end
+        end
+      end
+    end
+    if valid
+      push!(polys, PolyArea(newpolyrings))
+    end
+  end
+
+  n = length(polys)
+  out = n == 0 ? nothing : (n == 1 ? polys[1] : GeometrySet(polys))
+  return out
+end
+
+function clip(poly::Polygon, other::Geometry, method::WeilerAthertonClipping)
```

Comment:
could this work for any clipping method with a `Multi`? should we simply add a more general method in clipping.jl that handles multi using the `parent`?

Reply:
Moved this one as well. With slight change to sutherlandhodgeman.jl, seems to cooperate with it finely, but I did not see additional benefits to it in my tests, some malformed results were obtained. But then again the tests did use a bit more complex geometries that should not be expected to work with SutherlandHodgman. In theory, it might help with some non self-intersecting multi geometries, but couldn't confirm.